I updated other tests. It turns out that

> (implicitly assuming such attention module always have config attribute, otherwise it might suggest something is not propagating correctly to attn module/layer)

will fail for some classes, see below. I will use `getattr(..., None)` to pass those cases. 



#### Failing test.

```bash
FAILED tests/models/xlm_roberta_xl/test_modeling_xlm_roberta_xl.py::XLMRobertaXLModelTest::test_sdpa_can_dispatch_non_composite_models - AttributeError: 'XLMRobertaXLAttention' object has no attribute 'config'
FAILED tests/models/bert/test_modeling_bert.py::BertModelTest::test_sdpa_can_dispatch_non_composite_models - AttributeError: 'BertAttention' object has no attribute 'config'
FAILED tests/models/kosmos2/test_modeling_kosmos2.py::Kosmos2ModelTest::test_attn_implementation_composite_models - AttributeError: 'KosmosTextAttention' object has no attribute 'config'
FAILED tests/models/kosmos2/test_modeling_kosmos2.py::Kosmos2ModelTest::test_sdpa_can_dispatch_composite_models - AttributeError: 'KosmosTextAttention' object has no attribute 'config'
FAILED tests/models/vivit/test_modeling_vivit.py::VivitModelTest::test_sdpa_can_dispatch_non_composite_models - AttributeError: 'VivitAttention' object has no attribute 'config'
FAILED tests/models/vit_mae/test_modeling_vit_mae.py::ViTMAEModelTest::test_sdpa_can_dispatch_non_composite_models - AttributeError: 'ViTMAEAttention' object has no attribute 'config'
FAILED tests/models/videomae/test_modeling_videomae.py::VideoMAEModelTest::test_sdpa_can_dispatch_non_composite_models - AttributeError: 'VideoMAEAttention' object has no attribute 'config'
FAILED tests/models/dinov2_with_registers/test_modeling_dinov2_with_registers.py::Dinov2WithRegistersModelTest::test_sdpa_can_dispatch_non_composite_models - AttributeError: 'Dinov2WithRegistersAttention' object has no attribute 'config'
FAILED tests/models/yolos/test_modeling_yolos.py::YolosModelTest::test_sdpa_can_dispatch_non_composite_models - AttributeError: 'YolosAttention' object has no attribute 'config'
FAILED tests/models/beit/test_modeling_beit.py::BeitModelTest::test_sdpa_can_dispatch_non_composite_models - AttributeError: 'BeitAttention' object has no attribute 'config'
```

--- a/tests/test_modeling_common.py
+++ b/tests/test_modeling_common.py
@@ -4459,7 +4459,8 @@ class ModelTesterMixin:
                     for name, submodule in model_fa2.named_modules():
                         class_name = submodule.__class__.__name__
                         if (
-                            class_name.endswith("Attention")
+                            "Attention" in class_name
+                            and getattr(submodule, "config", None)
                             and submodule.config._attn_implementation == "flash_attention_2"
                         ):
                             has_fa2 = True

--- a/tests/test_modeling_common.py
+++ b/tests/test_modeling_common.py
@@ -3871,7 +3871,11 @@ class ModelTesterMixin:
 
             for name, submodule in model.named_modules():
                 class_name = submodule.__class__.__name__
-                if class_name.endswith("Attention") and submodule.config._attn_implementation != "eager":
+                if (
+                    class_name.endswith("Attention")
+                    and hasattr(submodule, "config")
+                    and submodule.config._attn_implementation != "eager"
+                ):
                     raise ValueError(
                         f"The eager model should not have SDPA/FA2 attention layers but got `{class_name}.config._attn_implementation={submodule.config._attn_implementation}`"
                     )
@@ -3905,7 +3909,11 @@ class ModelTesterMixin:
 
                 for name, submodule in model_eager.named_modules():
                     class_name = submodule.__class__.__name__
-                    if class_name.endswith("Attention") and submodule.config._attn_implementation == "sdpa":
+                    if (
+                        class_name.endswith("Attention")
+                        and hasattr(submodule, "config")
+                        and submodule.config._attn_implementation == "sdpa"
+                    ):
                         raise ValueError(
                             f"The eager model should not have SDPA attention layers but got `{class_name}.config._attn_implementation={submodule.config._attn_implementation}`"
                         )
@@ -3959,7 +3967,11 @@ class ModelTesterMixin:
 
                 for name, submodule in model_eager.named_modules():
                     class_name = submodule.__class__.__name__
-                    if class_name.endswith("Attention") and submodule.config._attn_implementation == "sdpa":
+                    if (
+                        class_name.endswith("Attention")
+                        and hasattr(submodule, "config")
+                        and submodule.config._attn_implementation == "sdpa"
+                    ):
                         raise ValueError("The eager model should not have SDPA attention layers")
 
     @parameterized.expand([("float16",), ("bfloat16",), ("float32",)])

--- a/tests/test_modeling_common.py
+++ b/tests/test_modeling_common.py
@@ -3873,7 +3873,7 @@ class ModelTesterMixin:
                 class_name = submodule.__class__.__name__
                 if (
                     class_name.endswith("Attention")
-                    and hasattr(submodule, "config")
+                    and getattr(submodule, "config", None)
                     and submodule.config._attn_implementation != "eager"
                 ):
                     raise ValueError(
@@ -3911,7 +3911,7 @@ class ModelTesterMixin:
                     class_name = submodule.__class__.__name__
                     if (
                         class_name.endswith("Attention")
-                        and hasattr(submodule, "config")
+                        and getattr(submodule, "config", None)
                         and submodule.config._attn_implementation == "sdpa"
                     ):
                         raise ValueError(
@@ -3969,7 +3969,7 @@ class ModelTesterMixin:
                     class_name = submodule.__class__.__name__
                     if (
                         class_name.endswith("Attention")
-                        and hasattr(submodule, "config")
+                        and getattr(submodule, "config", None)
                         and submodule.config._attn_implementation == "sdpa"
                     ):
                         raise ValueError("The eager model should not have SDPA attention layers")

